.eslintrc.yaml

```diff
@@ -169,6 +169,18 @@ rules:
   # see jest-community/eslint-plugin-jest#203 (rule might be removed?)
   jest/valid-describe: off
 
+  # The docs for this rule [1] say that placing an `expect` call
+  # outside a `test` or `it` block (e.g., in a `beforeEach` or
+  # `afterEach`) means that the assertion isn't executed. Empirically,
+  # this seems just wrong [2], and there are a few places where we
+  # want to call `expect` in a `beforeEach` or `afterEach` to assert
+  # that multiple tests in the same file aren't interfering with each
+  # other by leaving bits of state lying around.
+  #
+  # [1] https://github.com/jest-community/eslint-plugin-jest/blob/master/docs/rules/no-standalone-expect.md
+  # [2] https://github.com/zulip/zulip-mobile/pull/4235#discussion_r489984717
+  jest/no-standalone-expect: off
```

Comment:
Adapt to the first two and disable the third, which seems unhelpful,
    as noted in code comments. It did, however, lead to an improvement
    in `sentry-test`, where we move some `expect`s from `beforeEach` and
    `afterEach` into the tests themselves, since logically they're part
    of the tests themselves [1].

I do appreciate that these lint rules highlighted this improvement for us. But I think the credit can be assigned just as well to `jest/expect-expect`, which we're keeping :slightly_smiling_face: . The two tests that were improved were tests that had no `expect` calls of their own.

In fact I think it's not a coincidence that the tests that were improved were tests that would have been flagged by `expect-expect` alone, while the tests that wouldn't have been improved were tests that that rule thought was fine and only `no-standalone-expect` complained about. The fact that those Sentry tests had no `expect` of their own is closely tied to the fact that they were relying on `afterEach` to do the actual testing; and the fact that the heartbeat tests were more self-contained tests and their `afterEach` was more about protecting the tests from each other is closely tied to the fact that those tests had their own `expect` calls where they did their job as tests.

Reply:
Makes sense!